mne/io/base.py

```diff
@@ -1386,7 +1386,10 @@ def resample(
         sfreq = float(sfreq)
         o_sfreq = float(self.info["sfreq"])
         if _check_resamp_noop(sfreq, o_sfreq):
-            return self
+            if events:
+                return self, events
```

Comment:
if `events` is an ndarray this is going to be a problem. So better:
```suggestion
            if events is not None:
                return self, events.copy()
```
and the `copy` added since it's always copied below